x-pack/solutions/security/plugins/security_solution/public/cases/components/ai_for_soc/wrapper.tsx

```diff
@@ -0,0 +1,127 @@
+import {
+  type EuiDataGridColumn,
+  EuiEmptyPrompt,
+  EuiSkeletonLoading,
+  EuiSkeletonRectangle,
+} from '@elastic/eui';
+import type { QueryDslQueryContainer } from '@elastic/elasticsearch/lib/api/types';
+import { i18n } from '@kbn/i18n';
+import type { Alert } from '@kbn/alerting-types';
+import { Table } from './table';
+import { useFetchIntegrations } from '../../../detections/hooks/alert_summary/use_fetch_integrations';
+import { useFindRulesQuery } from '../../../detection_engine/rule_management/api/hooks/use_find_rules_query';
+import { useKibana } from '../../../common/lib/kibana';
+
+const DATAVIEW_ERROR = i18n.translate(
+  'xpack.securitySolution.attackDiscovery.aiForSocTableTab.dataViewError',
+  {
+    defaultMessage: 'Unable to create data view',
+  }
+);
+
+export const LOADING_PROMPT_TEST_ID = 'cases-alert-loading-prompt';
+export const ERROR_TEST_ID = 'cases-alert-error';
+export const SKELETON_TEST_ID = 'cases-alert-skeleton';
+export const CONTENT_TEST_ID = 'cases-alert-content';
+
+const dataViewSpec: DataViewSpec = { title: '.alerts-security.alerts-default' };
+
+interface AiForSOCAlertsTableProps {
+  /**
+   * Id to pass down to the ResponseOps alerts table
+   */
+  id: string;
+  /**
+   * Callback fired when the alerts have been first loaded
+   */
+  onLoaded?: (alerts: Alert[], columns: EuiDataGridColumn[]) => void;
+  /**
+   * Query that contains the id of the alerts to display in the table
+   */
+  query: Pick<QueryDslQueryContainer, 'bool' | 'ids'>;
+}
+
+/**
+ * Component used in the Cases page under the Alerts tab, only in the AI4DSOC tier.
+ * It fetches rules, packages (integrations) and creates a local dataView.
+ * It renders a loading skeleton while packages are being fetched and while the dataView is being created.
+ */
+export const AiForSOCAlertsTable = memo(({ id, onLoaded, query }: AiForSOCAlertsTableProps) => {
+  const { data } = useKibana().services;
+  const [dataView, setDataView] = useState<DataView | undefined>(undefined);
+  const [dataViewLoading, setDataViewLoading] = useState<boolean>(true);
+
+  // Fetch all integrations
+  const { installedPackages, isLoading: integrationIsLoading } = useFetchIntegrations();
+
+  // Fetch all rules. For the AI for SOC effort, there should only be one rule per integration (which means for now 5-6 rules total)
+  const { data: ruleData, isLoading: ruleIsLoading } = useFindRulesQuery({});
+  const ruleResponse = useMemo(
+    () => ({
+      rules: ruleData?.rules || [],
+      isLoading: ruleIsLoading,
+    }),
+    [ruleData, ruleIsLoading]
+  );
+
+  useEffect(() => {
+    let dv: DataView;
+    const createDataView = async () => {
+      dv = await data.dataViews.create(dataViewSpec);
+      setDataView(dv);
+      setDataViewLoading(false);
+    };
+    createDataView();
+
+    // clearing after leaving the page
+    return () => {
+      if (dv?.id) {
+        data.dataViews.clearInstanceCache(dv?.id);
```

Comment:
nit: 
this `?.` generates an unnecessary condition in the build while in a block where it's always present

```suggestion
        data.dataViews.clearInstanceCache(dv.id);
```

Reply:
That is correct, thanks for catching that. Removed!